opentelemetry-http/src/lib.rs

```diff
@@ -105,21 +105,23 @@ pub mod hyper {
 
     use super::{async_trait, Bytes, HttpClient, HttpError, Request, Response};
     use http::HeaderValue;
-    use hyper::client::connect::Connect;
-    use hyper::Client;
+    use http_body_util::BodyExt;
+    use hyper::body::Body;
+    use hyper_util::client::legacy::{connect::Connect, Client};
+    use std::error::Error;
     use std::fmt::Debug;
     use std::time::Duration;
     use tokio::time;
 
     #[derive(Debug, Clone)]
-    pub struct HyperClient<C> {
-        inner: Client<C>,
+    pub struct HyperClient<C, B> {
```

Comment:
BTW, have you considered abstracting over `B` here? Like, picking some concrete type (maybe vendoring `http_body_util::Full`) instead of exposing this as a type parameter on `HypercClient`?

Reply:
What about this solution I just pushed?
I like the idea of this wrapper more than vendoring anything.

Reply:
Yup, that's probably better in this case! One nit: that file is laid out in top-down order, so would be better if the `Body` wrapper moves down to somewhere near the bottom of the module (probably before `ResponseExt`).

Reply:
Even though we leak some information about the inner impl that could *technically* cause some headaches if it's ever swapped out for something else.
The `Infallible` type. Maybe this should become a box dyn error?

Reply:
Yeah, that seems reasonable.